ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldes/retentionpolicy/creation/TimeBasedRetentionPolicyCreator.java

```diff
@@ -0,0 +1,41 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.creation;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.converter.DurationParser;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.converter.RdfModelConverter;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.RetentionPolicy;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.timebased.TimeBasedRetentionPolicy;
+import org.apache.jena.datatypes.RDFDatatype;
+import org.apache.jena.datatypes.xsd.XSDDuration;
+import org.apache.jena.rdf.model.*;
+import org.apache.jena.rdf.model.impl.LiteralImpl;
+import org.apache.jena.riot.Lang;
+
+import java.time.Duration;
+import java.util.List;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.domain.constants.RdfConstants.TREE;
+import static org.apache.jena.rdf.model.ResourceFactory.createProperty;
+
+public class TimeBasedRetentionPolicyCreator implements RetentionPolicyCreator {
+	public static final Property TREE_VALUE = createProperty(TREE, "value");
+
+	@Override
+	public RetentionPolicy createRetentionPolicy(Model model) {
+		List<Statement> treeValueStatements = model.listStatements(null, TREE_VALUE, (RDFNode) null).toList();
+		if (treeValueStatements.size() != 1) {
+			throw new IllegalArgumentException(
+					"Cannot Create Time Based Retention Policy in which there is not exactly 1 " + TREE_VALUE.toString()
+							+ " statement.\n Found " + treeValueStatements.size() + " statements in :\n"
+							+ RdfModelConverter.toString(model, Lang.TURTLE));
+		}
```

Comment:
not so much a remark, rather a question. COULD we add a custom exception for this? Would it be useful ?

Reply:
Will keep it in mind. Probably we would like to have one exception for all the possible failures during the creations of retention policies. Because, we would like to convey that exception to the end-user so that he/she can update what is needed. And in that way we would only have to translate one exception in the controller. Otherwise, we risk having too many exceptions which all need their translation in the controller (information is leaking).

Will create something to catch all retention policy creation exceptions when I work on the other policies.